core/data/compare/comparator.go

```diff
@@ -152,18 +154,20 @@ func (t Comparator) compareValues(v1, v2 reflect.Value, ptr bool) {
 		t.compareValues(v1.Elem(), v2.Elem(), true)
 	case reflect.Struct:
 		t1 := v1.Type()
-		hidden := false
 		for i, n := 0, t1.NumField(); i < n; i++ {
 			f := t1.Field(i)
 			if r, _ := utf8.DecodeRuneInString(f.Name); unicode.IsUpper(r) {
 				t.With(t.Path.Member(f.Name, toValue(v1, ptr), toValue(v2, ptr))).compareValues(v1.Field(i), v2.Field(i), false)
 			} else {
-				hidden = true
+				// Filthy hack to inspect hidden fields.
```

Comment:
Neat, but also terrifying.

Reply:
No production code should be using this, so I deemed it acceptable.